src/app/clusters/group-key-mgmt-server/group-key-mgmt-server.cpp

```diff
@@ -0,0 +1,410 @@
+        printStatus();
+        return EMBER_ZCL_STATUS_SUCCESS;
+    }
+    else
+    {
+        return EMBER_ZCL_STATUS_NOT_FOUND;
+    }
+}
+
+static EmberAfStatus revokeKey(uint16_t fabricIndex, GroupId groupId, uint16_t keySetIndex)
+{
+    sGroupStates.erase(std::tuple<uint16_t, GroupId, uint16_t>(fabricIndex, groupId, keySetIndex));
+    printStatus();
+    return EMBER_ZCL_STATUS_SUCCESS;
+}
+
+static EmberAfStatus revokeKeys(uint16_t fabricIndex, uint16_t keySetIndex)
+{
+    std::set<std::tuple<uint16_t, GroupId, uint16_t>>::iterator it = sGroupStates.begin();
+    while (it != sGroupStates.end())
+    {
+        auto curr = it++;
+        if (std::get<0>(*curr) == fabricIndex && std::get<2>(*curr) == keySetIndex)
+        {
+            sGroupStates.erase(curr);
+        }
+    }
+    printStatus();
+    return EMBER_ZCL_STATUS_SUCCESS;
+}
+
+static EmberAfStatus revokeKeys(uint16_t fabricIndex)
+{
+    std::set<std::tuple<uint16_t, GroupId, uint16_t>>::iterator it = sGroupStates.begin();
+    while (it != sGroupStates.end())
+    {
+        auto curr = it++;
+        if (std::get<0>(*curr) == fabricIndex)
+        {
+            sGroupStates.erase(curr);
+        }
+    }
+    printStatus();
+    return EMBER_ZCL_STATUS_SUCCESS;
+}
+
+//------------------------------------------------------------------------------
+// Group Key Management Cluster server callbacks
+//------------------------------------------------------------------------------
+
+/** @brief Group Key Management Cluster Server Init
+ *
+ * Server Init
+ *
+ * @param endpoint    Endpoint that is being initialized
+ */
+void emberAfGroupKeyManagementClusterServerInitCallback(chip::EndpointId endpoint) {}
+
+/**
+ * @brief  Cluster AddKey Command callback (from client)
+ */
+bool emberAfGroupKeyManagementClusterSetKeyCallback(chip::EndpointId endpoint, chip::app::CommandHandler * commandObj,
+                                                    uint16_t fabricIndex, uint16_t keySetIndex, chip::ByteSpan keyRoot,
+                                                    chip::ByteSpan epochStartTime, uint8_t securityPolicy)
+{
+    EmberAfStatus result = EMBER_ZCL_STATUS_FAILURE;
+
+    if (0 == endpoint)
+    {
+        result = setKey(fabricIndex, keySetIndex, keyRoot, epochStartTime, securityPolicy);
+    }
+    else
+    {
+        result = EMBER_ZCL_STATUS_INVALID_VALUE;
+    }
+
+    EmberStatus status = emberAfSendImmediateDefaultResponse(result);
+    if (EMBER_SUCCESS != status)
+    {
+        emberAfGroupsClusterPrintln("Group Key Management: failed to send %s: 0x%x", "default_response", status);
+    }
+    return true;
+}
+
+/**
+ * @brief  Cluster RemoveKey Command callback (from client)
+ */
+bool emberAfGroupKeyManagementClusterRemoveKeyCallback(chip::EndpointId endpoint, chip::app::CommandHandler * commandObj,
+                                                       uint16_t fabricIndex, uint16_t keySetIndex)
```

Comment:
The fabricIndex for all these sample commands (and eventually, the attributes) will have to be checked against the incoming session's fabric index. Otherwise, it would be possible for one admin to mutate/access another fabric's keys, which is forbidden. Suggest adding TODOs about making sure we do not allow access of one fabric's keys from another fabric.